File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.Duration;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.temporal.ChronoUnit;
import java.util.Optional;

import static java.util.Objects.requireNonNull;

class Gigasecond {
//    private static final long GIGASECOND = (long) Math.pow(10, 9);
//
//    private LocalDateTime birthDateTime;
//
//    Gigasecond(LocalDate birthDate) {
//        this.birthDateTime = birthDate.atStartOfDay();
//    }
//
//    Gigasecond(LocalDateTime birthDateTime) {
//        this.birthDateTime = birthDateTime;
//    }
//
//    LocalDateTime getDate() {
//        return birthDateTime.plus(Duration.ofSeconds((int)1e9));
//    }

    static final long SECONDS = 1_000_000_000;

    private LocalDateTime localDateTime;

    Gigasecond(LocalDate birthDate) {
            this(birthDate.atStartOfDay());
    }

    Gigasecond(LocalDateTime birthDateTime) {
        this.localDateTime = addSeconds(birthDateTime, SECONDS);
    }

    LocalDateTime addSeconds(LocalDateTime birthDate, long seconds) {
        return birthDate.plus(seconds, ChronoUnit.SECONDS);
    }

    LocalDateTime getDate() {
        return this.localDateTime;
    }

}

File: csharp/gigasecond/Gigasecond.cs
using System;

public static class Gigasecond
{
    private static readonly int NO_HOURS   = 0;
    private static readonly int NO_MINUTES = 0;
    private static readonly int GIGA_SECONDS = Convert.ToInt32(Math.Pow(10, 9));
    private static readonly TimeSpan ONE_GIGASECOND = new TimeSpan(NO_HOURS, NO_MINUTES, GIGA_SECONDS);

    public static DateTime Add(DateTime birthDate) => birthDate.Add(ONE_GIGASECOND);
}